toolbox/bulk_processing/validators.py

```diff
@@ -123,3 +125,23 @@ def validate(expected_capacity, **kwargs):
                 f'CE Unit Expected Capacity "{expected_capacity}" cannot be null, blank or zero')
 
     return validate
+
+
+def ce_e_has_expected_capacity():
+    def validate(expected_capacity, **kwargs):
+        if kwargs['row']['ADDRESS_TYPE'] == 'CE' and kwargs['row']['ADDRESS_LEVEL'] == 'E' and (
+                kwargs['row']['TREATMENT_CODE'] not in {'CE_LDCEE', 'CE_LDCEW'}) and (
+                not expected_capacity.isdigit() or int(expected_capacity) == 0):
+            raise Invalid(
+                f'Expected Capacity "{expected_capacity}" must be greater than 0')
+
+    return validate
+
+
+def alphanumeric_postcode():
+    def validate(postcode, **_kwargs):
+        postcode = postcode.replace(" ", "")
+        if not postcode.isalnum():
+            raise Invalid(f'Postcode "{postcode}" is non alphanumeric')
```

Comment:
This will spit out the postcode with whitespace stripped, it should present the exact invalid value back.